wts/blueprints/tokens.py

```diff
@@ -18,4 +19,12 @@ def get_token():
             expires = int(expires)
         except ValueError:
             return flask.jsonify({"error": "expires has to be an integer"}), 400
-    return flask.jsonify({"token": get_access_token(expires=expires)})
+
+    requested_idp = flask.request.args.get("idp", "default")
+    try:
+        client = get_oauth_client(idp=requested_idp)
+    except KeyError:
+        return flask.jsonify({"error": "requested idp not configured"}), 400
```

Comment:
i don't think we need this try/catch, we should patch `get_oauth_client` to throw a `UserError` instead of a `KeyError` - then the error handler should automatically return 400 for all endpoints instead of just this one

and then we can move `requested_idp` and `client` back to `get_access_token`

Reply:
I'll patch `get_oath_client` to throw UserError, but I like passing in the arguments better than having a helper function that relies on some flask state.
Why is it better to have `get_oath_token()` harvest its inputs from global flask state rather than pass in the inputs as arguments?  The set some global state, then call a function that accesses that state as a side effect thing makes me itch.

Reply:
ugh - never mind